src/app/shared/components/template/services/template-variables.service.ts

```diff
@@ -365,11 +372,11 @@ export class TemplateVariablesService extends AsyncServiceBase {
         parsedValue = evaluateJSExpression(expression, thisCtxt, globalFunctions, globalConstants);
         break;
       case "item":
-        log("evaluate item", evaluator, context);
-        try {
+        // only attempt to evaluate items if context passed, otherwise leave as original unparsed string
```

Comment:
Yes that is correct, we have both the item data list and template begin/end data_items rows, and create a loop for each data list row (termed `itemContext`), and evaluate the template rows replacing references to `@item` with that context variable.

The catch here is because data_items are processed after being rendered, we need to make sure the parser ignores references to `@item` and leaves for the data-items component to handle with the appropriate context.

This is different than the original begin/end items, which has a hardcoded intervention in the parser to process directly instead of delegating to the child component (retrieving the item context itself). The benefit to this approach is that we don't need to render an additional container component for the data_items, the downside is we would have to re-render/reprocess at the template level any time the list changes (to force full reprocess), instead of just letting the items render once and manage their own change detection (which is what data_items do)